tests/nlu/selectors/test_selectors.py

```diff
@@ -268,105 +294,176 @@ async def test_train_model_checkpointing(
     assert len(all_files) > 4
 
 
-async def _train_persist_load_with_different_settings(
+def _train_persist_load_with_different_settings(
     pipeline: List[Dict[Text, Any]],
-    component_builder: ComponentBuilder,
-    tmp_path: Path,
+    config_params: Dict[Text, Any],
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
     should_finetune: bool,
 ):
-    _config = RasaNLUModelConfig({"pipeline": pipeline, "language": "en"})
+    loaded_pipeline = [
+        registry.get_component_class(component.pop("name"))(component)
+        for component in copy.deepcopy(pipeline)
+    ]
 
-    (trainer, trained, persisted_path) = await rasa.nlu.train.train(
-        _config,
-        path=str(tmp_path),
-        data="data/examples/rasa/demo-rasa.yml",
-        component_builder=component_builder,
+    importer = RasaFileImporter(
+        training_data_paths=["data/examples/rasa/demo-rasa.yml"]
+    )
+    training_data = importer.get_nlu_data()
+
+    for component in loaded_pipeline:
+        component.train(training_data)
+
+    if should_finetune:
+        default_execution_context.is_finetuning = True
```

Comment:
this was done after the training. Now you're doing it _before_ the training which makes it fail.

Finetuning takes an already trained model and improves it. So instead of starting the training from scratch, it uses an already trained model and somewhat "continues" training. Which is why you're getting the exception: You can't finetune without having a trained model first. Does that make sense?

Reply:
Got it, perfect TIL 😁 ! Reshuffled this after the training bit and it passed now.